tests/test_plugins/test_adjoint.py

```diff
@@ -1563,27 +1563,41 @@ def make_custom_medium(num_cells: int) -> JaxCustomMedium:
     assert sim == sim2
 
 
-def test_num_input_structures(use_emulated_run, tmp_path):
+def test_num_input_structures(use_emulated_run, tmp_path, monkeypatch):
     """Assert proper error is raised if number of input structures is too large."""
 
-    def make_sim_(num_input_structures: int) -> JaxSimulation:
-        sim = make_sim(permittivity=EPS, size=SIZE, vertices=VERTICES, base_eps_val=BASE_EPS_VAL)
+    test_max_num_structs = 3  # monkeypatch for easier testing
+    monkeypatch.setattr(simulation, "MAX_NUM_INPUT_STRUCTURES", test_max_num_structs)
+
+    def make_sim_(permittivity=EPS, num_input_structures: int = 1) -> JaxSimulation:
+        sim = make_sim(
+            permittivity=permittivity, size=SIZE, vertices=VERTICES, base_eps_val=BASE_EPS_VAL
+        )
         struct = sim.input_structures[0]
         return sim.updated_copy(input_structures=num_input_structures * [struct])
 
-    sim = make_sim_(num_input_structures=MAX_NUM_INPUT_STRUCTURES)
+    sim = make_sim_(num_input_structures=test_max_num_structs)
     sim._validate_web_adjoint()
 
-    sim = make_sim_(num_input_structures=MAX_NUM_INPUT_STRUCTURES + 1)
+    sim = make_sim_(num_input_structures=test_max_num_structs + 1)
     with pytest.raises(AdjointError):
         sim._validate_web_adjoint()
 
     # make sure that the remote web API fails whereas the local one passes
     with pytest.raises(AdjointError):
         sim_data = run(sim, task_name="test", path=str(tmp_path / RUN_FILE))
 
+    # no error when calling run_local directly
     sim_data = run_local(sim, task_name="test", path=str(tmp_path / RUN_FILE))
 
+    # no error when calling it inside a gradient computation
+    def f(permittivity):
+        sim = make_sim_(permittivity=permittivity, num_input_structures=test_max_num_structs + 1)
+        sim_data = run_local(sim, task_name="test", path=str(tmp_path / RUN_FILE))
+        return objective(extract_amp(sim_data))
+
+    _ = grad(f)(EPS)
+
```

Comment:
let's throw in a test asserting that AdjointError is raised if `grad ` is called on a function where `run` is used? for good measure? unless this seems unnecessary